### PR TITLE
Followup to menu refactor. TMC warnings, sanity.

### PR DESCRIPTION
**Pull request by Ludy87**
» unused variables `index` https://github.com/MarlinFirmware/Marlin/issues/12281
» error: 'zprobe_zoffset' was not declared in this scope
» ADC_KEYPAD & REPRAPWORLD_KEYPAD https://github.com/MarlinFirmware/Marlin/issues/12285
» Check existing RX/TX pins against enable TMC UART drivers.

([Concise Diff](12288/files?w=1))

### CODE REVIEW

**Comment by Ludy87** [comment]
@thinkyhead Sorry for my bad fast fixes

**Comment by thinkyhead** [comment]
It's all good. I appreciate having any oversights pointed out. My own testing and Travis CI are easily able to miss things because not every combination of options can be tested.

**Comment by dahuafschmied** [comment]
just a question. Is it really necessary to define RX and TX Pins even if i use hardware serial (3)? Serial communication is functional even without defined pins.
if it has to be please add a better comment/description about hardware/software serial to the config .h to avoid confusion.

**Comment by Ludy87** [comment]
You're right, I did not think that.

File: Marlin/src/gcode/feature/trinamic/M911-M915.cpp [hunk 1]
```diff
@@ -271,7 +304,7 @@
 /**
  * M914: Set StallGuard sensitivity.
  */
-#if USE_SENSORLESS
+#if USE_SENSORLESS && (X_SENSORLESS || Y_SENSORLESS || Z_SENSORLESS)
```
Comment (teemuatlut) [inline Marlin/src/gcode/feature/trinamic/M911-M915.cpp]: Is there a case where a user would enable sensorless homing/probing but then uncomment all stall sensitivities?

Comment (Ludy87) [reply Marlin/src/gcode/feature/trinamic/M911-M915.cpp]: this Problem is, where E0, E1, E2 ...  define as TMC2130 and Sensorless give it a unses var warning

There are stupid users like me :-P

Comment (teemuatlut) [reply Marlin/src/gcode/feature/trinamic/M911-M915.cpp]: Then that's a configuration error and should rather be handled in sanitycheck.

Comment (Ludy87) [reply Marlin/src/gcode/feature/trinamic/M911-M915.cpp]: `const uint8_t index = parser.byteval ('I');`
It would still be unused.

Comment (Ludy87) [reply Marlin/src/gcode/feature/trinamic/M911-M915.cpp]: Or do you mean the check'if USE_SENSORLESS && (X_SENSORLESS || Y_SENSORLESS || Z_SENSORLESS) `in` SanityCheck.h`

Comment (teemuatlut) [reply Marlin/src/gcode/feature/trinamic/M911-M915.cpp]: `const uint8_t index` requires `USE_SENSORLESS` to be true.
Sanitycheck can force this to be the case only when an applicable driver is enabled on XYZ.

Comment (Ludy87) [reply Marlin/src/gcode/feature/trinamic/M911-M915.cpp]: > const uint8_t index requires USE_SENSORLESS to be true.

Yes, but if the E0 driver is TMC2130 and XYZ is not, then `index` is unused

Comment (teemuatlut) [reply Marlin/src/gcode/feature/trinamic/M911-M915.cpp]: If E0 is TMC2130 and XYZ is not **and** sensorless homing/probing is **enabled**, you get a sanity check error.

If E0 is TMC2130 and XYZ is not **and** sensorless homing/probing is **disabled**, the code never gets compiled in.

Comment (Ludy87) [reply Marlin/src/gcode/feature/trinamic/M911-M915.cpp]: #if USE_SENSORLESS  && !(X_SENSORLESS || Y_SENSORLESS || Z_SENSORLESS)
      #error "SENSORLESS_HOMING or SENSORLESS_PROBING is enabled, requires TMC2130 for X or Y or Z"
    #endif

**Comment by thinkyhead** [comment]
Rebased, squashed, adjusted sanity checks and keypad defines.
```
git fetch origin
git checkout bugfix-2.0.x-german
git reset --hard origin/bugfix-2.0.x-german
```

**Pull request merged**